src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -427,22 +446,21 @@ export class DashboardAppController {
               refreshDashboardContainer();
             });
 
-            const incomingState = embeddable
-              .getStateTransfer(scopedHistory())
-              .getIncomingEmbeddablePackage();
-            if (incomingState) {
-              if ('id' in incomingState) {
-                container.addOrUpdateEmbeddable<SavedObjectEmbeddableInput>(incomingState.type, {
-                  savedObjectId: incomingState.id,
-                });
-              } else if ('input' in incomingState) {
-                const input = incomingState.input;
-                delete input.id;
-                const explicitInput = {
-                  savedVis: input,
-                };
-                container.addOrUpdateEmbeddable<EmbeddableInput>(incomingState.type, explicitInput);
```

Comment:
can you please explain this changes ? why was this necessary, seems we did addOrUpdate before and now we update here directly and only use container.add  ?

Reply:
This is a strange one, and just a slightly better workaround to https://github.com/elastic/kibana/issues/70381 than the `addOrUpdate` code was.

Basically, due to the embeddable panels not updating properly to changes in their input, we have to find some other way to reflect the changes from the editor after fetching them from the `embeddableStateTransfer`. the `addOrUpdate` code looked a little simpler, but behind the scenes it actually required the old panel to be loaded, a new embeddableID to be generated, the old panel to be deleted and a new one to be put in its place.

I've implemented a new workaround by changing the panel's input before the dashboard input is loaded here:
https://github.com/elastic/kibana/blob/cd61dfc7985e9d16c53d465f1aa093034891823d/src/plugins/dashboard/public/application/dashboard_app_controller.tsx#L315-L327
This allows us to retain the same `embeddableId`, as well as avoid loading the panel to immediately replace it.

Because the 'update' case is already handled by the above snippet, we can call `addNewEmbeddable` here.